src/NC-State-Parks.ino

```diff
@@ -279,16 +280,49 @@ void setup()                                        // Note: Disconnected Setup(
 void loop()
 {
   switch(state) {
+  case PARTICLE_CONNECT_STATE:                                          // Will currently always come from REPORTING_STATE when not connected
+    if (!waitingForConnection) {                                        // Must be just entering this state -> should only be run once when connecting
+      connectionStartTime = millis();
+      Cellular.on();                                                    // Needed until they fix this: https://github.com/particle-iot/device-os/issues/1631
+      Particle.connect();
+      waitingForConnection = true;                                      // Start waiting for connection
+    }
+    if (Particle.connected()) {                                         // Successfully connected!
+      char connectionStr[32];
+      sysStatus.connectedStatus = true;
+      sysStatus.lastConnection = Time.now();
+      snprintf(connectionStr, sizeof(connectionStr),"Connected in %lu secs", millis()-connectionStartTime);
+      Log.info(connectionStr);
+      if (sysStatus.verboseMode) publishQueue.publish("Cellular",connectionStr,PRIVATE);
+      systemStatusWriteNeeded = true;
+      waitingForConnection = false;                                     // No longer waiting for connection
+      Particle.syncTime();
+      state = REPORTING_STATE;                                          // We can go back to reporting state now that we have connected
+    }
+    else if ((millis()-connectionStartTime) > connectionTimeout) {      // Time has run out
+      waitingForConnection = false;                                     // Given up waiting for connection
+      sysStatus.connectedStatus = false;
+      Log.info("cloud connection unsuccessful");
+      fram.put(FRAM::systemStatusAddr,sysStatus);
+      Log.warn("failed to connect to cloud, doing deep reset");
+      delay(100);
+      ab1805.deepPowerDown();
+      systemStatusWriteNeeded = true;
```

Comment:
None of the lines after deepPowerDown will be executed as the device will reset and go back to setup()

Reply:
Would you rather the `deepPowerDown` be done in the `ERROR_STATE` and not here? Then, I will just transition to the `ERROR_STATE`.